tests/agent_unittests/test_agent_protocol.py

```diff
@@ -522,7 +522,8 @@ def test_audit_logging():
     protocol = AgentProtocol(settings, client_cls=HttpClientRecorder)
     protocol.send("preconnect")
 
-    with Path(f.name).open() as f:
+    audit_log_path = Path(f.name)
+    with audit_log_path.open() as f:
```

Comment:
Should this have an encoding on it?

Reply:
No, the audit log file is being written to in the actual agent code in the [preferred encoding](https://docs.python.org/3/library/functions.html#open) as returned by `locale.getencoding()`. We read it back in using the same default encoding.

The other tests that needed encoding specified are pre-written files from Mac/Linux machines that encode unicode characters in UTF-8, which is not the default for Windows and produces errors.